src/main/java/org/apache/activemq/broker/scheduler/SchedulerUtils.java

```diff
@@ -93,8 +95,22 @@ public Message createMessage(long start, long end) throws Exception {
 		return message;
 	}
 
-	public Message toMessage(String id, ByteSequence payload) throws Exception {
+	public ByteSequence toByteSequence(Message message) throws Exception {
+		return wireFormat.marshal(message);
+	}
+	public Message toMessage(ByteSequence payload) throws Exception {
 		Message message = (Message)wireFormat.unmarshal(payload);
+
+		return message;
+	}
+	public Message copyProperties(Message toMessageJob, Message fromMessage) throws Exception {
+		for (Map.Entry<String,Object> entry : fromMessage.getProperties().entrySet()) {
+			toMessageJob.setProperty(entry.getKey(), entry.getValue());
+		}
+		return toMessageJob;
```

Comment:
There is no reason to return a message from copyProperties, as the properties are copied into toMessageJob, which we already have access to.